Core/src/IfcGeometryConverter/GeomUtils.h

```diff
@@ -62,93 +62,286 @@ namespace OpenInfraPlatform
 					buw::Vector3f& closest);
 
+				/**********************************************************************************************/
+				/*! \brief Verify if point is on the segment of the line. 
+				*
+				* \param[in] lambda					Lambda of the line. 
+				* \param[in] point					Point which should be verified. 
+				* \param[in] line_origin			Line origin point.
+				* \param[in] line_direction			Direction of the line.
+				*
+				* \return 							Returns true, if point is on the line. False otherwise.
+				*/
 				static bool isPointOnLineSegment(double& lambda,
 					const buw::Vector3f& point,
 					const buw::Vector3f& line_origin,
 					const buw::Vector3f& line_direction);
-
+				
+				/**********************************************************************************************/
+				/*! \brief Extrude geometry. 
+				*
+				* \param[in] paths					List of several sets of points of the geometry.
+				* \param[in] dir					The direction, where geometry should be extruded.
+				* \param[in] closed					Boolean operator, which shows, if geometry is closed.
+				*
+				* \param[out] poly_data				List of faces of the geometry.
+				*/
 				static void extrude(const std::vector<std::vector<carve::geom::vector<2>>>& paths,
 					const carve::geom::vector<3> dir,
 					const bool closed,
 					std::shared_ptr<carve::input::PolyhedronData>& poly_data);
 
+				/**********************************************************************************************/
+				/*! \brief Incorporates voids to the geometry. 
+				*
+				* \param[in] paths					List of several sets of points. 
+				*
+				* \param[out] merged_path			Arranged list of points.
+				* \param[out] triangulated			List of triangles to visualize geometry.
+				* \param[out] path_all_loops		List of several sets of indices. 
+				*/
 				static void incorporateVoids(const std::vector<std::vector<carve::geom2d::P2>>& paths,
 					std::vector<carve::geom2d::P2>& merged_path,
 					std::vector<carve::triangulate::tri_idx>& triangulated, 
 					std::vector<std::pair<size_t, size_t>>& path_all_loops);
 
+				/**********************************************************************************************/
+				/*! \brief Corrects a winding of the points in curves.
+				*
+				* \param[in] face_loops_input		List of several sets of points.
+				*
+				* \param[out] normal_first_loop		Normalised first loop. 
+				*
+				* return							List of several sets of points with correct winding. 			
+				*/
 				static std::vector<std::vector<carve::geom2d::P2>> GeomUtils::correctWinding(
 					const std::vector<std::vector<carve::geom::vector<2>>> & face_loops_input, 
 					carve::geom::vector<3>& normal_first_loop);
 
+				/**********************************************************************************************/
+				/*! \brief ???
+				*
+				* \param[in] eye					???
+				* \param[in] center					???
+				* \param[in] up						??? 
+				*
+				* \param[out] m						???
+				*/

```

Comment:
I do not know, how to describe this function. Can you please give me a hint?

Reply:
That's a bummer. I don't know that exactly either. We'll have to look where this function is being used, perhaps that will open our eyes.